src/routes/+layout.ts

```diff
@@ -1,25 +1,44 @@
 // export const ssr = false
 // export const prerender = false
-import { NineBallGame, Player, NineBallRack } from '$lib';
+import { NineBallGame, NineBallPlayer, NineBallRack } from '$lib';
+import { EightBallGame, EightBallRack, EightBallPlayer } from '$lib/eight-ball';
 import type { ToastProps, DialogProps } from '$lib/components';
 import { persisted } from 'svelte-persisted-store';
 import { writable } from 'svelte/store';
 
 export function load() {
 	const dialog = persisted<DialogProps | null>('dialog', null, { storage: 'session' });
 	const toastTime = writable<number>(0);
-	const game = persisted<NineBallGame | null>('game', null, {
+	const nineBallGame = persisted<NineBallGame | null>('game', null, {
 		storage: 'session',
 		serializer: {
 			parse: (str) => {
 				const game: NineBallGame = Object.setPrototypeOf(JSON.parse(str), NineBallGame.prototype);
-				game.players.forEach((player: Player) => Object.setPrototypeOf(player, Player.prototype));
+				game.players.forEach((player: EightBallPlayer) =>
+					Object.setPrototypeOf(player, EightBallPlayer.prototype)
+				);
+				game.racks.forEach((rack) => Object.setPrototypeOf(rack, EightBallRack.prototype));
+				return game;
+			},
+			stringify: JSON.stringify
+		}
+	});
+	const eightBallGame = persisted<EightBallGame | null>('game', null, {
```

Comment:
you can move these into individual routes. I.e. 
`eighBallGame` to `eight-ball/+page.ts`

```typescript
return {
         game: persisted<EightBallGame | null>('game', null, {
		storage: 'session',
		serializer: {
			parse: (str) => {
				const game: EightBallGame = Object.setPrototypeOf(JSON.parse(str), EightBallGame.prototype);
				game.players.forEach((player: NineBallPlayer) =>
					Object.setPrototypeOf(player, NineBallPlayer.prototype)
				);
				game.racks.forEach((rack) => Object.setPrototypeOf(rack, NineBallRack.prototype));
				return game;
			},
			stringify: JSON.stringify
		}
	});
}
```